zeeguu/core/word_scheduling/basicSR/basicSR.py

```diff
@@ -116,9 +143,63 @@ def find_or_create(cls, db_session, bookmark):
         db_session.commit()
         return b
 
+    @classmethod
+    def priority_bookmarks_to_study(cls, user, required_count):
+        """
+        Prioritizes the bookmarks to study. To randomize the
+        exercise order utilize the Frontend assignBookmarksToExercises.js
+
+        The original logic is kept in bookmarks_to_study as it is called to
+        get similar_words to function as distractors in the exercises.
+
+        Currently, we prioritize bookmarks in the following way:
+        1. Words that are closest to being learned (indicated by `cooling_interval`, the highest the closest it is)
+        2. Words that are most common in the language (utilizing the word rank in the db)
+        """
+
+        def sorting_properties(bookmark):
+            cooling_interval = (
+                cls.query.filter_by(bookmark=bookmark).one().cooling_interval
+            )
+            user_word = UserWord.query.filter_by(id=bookmark.origin_id).one()
+            word_rank = user_word.rank
+            if word_rank is None:
+                word_rank = 10000
+            return (-cooling_interval, word_rank)
+
+        tomorrow = cls.get_current_study_window()
```

Comment:
`end_of_today` is slightly more precise than tomorrow? 
Same question elsewhere where `tomorrow` is being used.